### ISSUE TITLE
Using TLS with Consul discovery is broken

### ISSUE THREAD

**Issue by technolo-g** [docker/swarm#226]
Currently when attempting to use the Consul discovery mechanism there is no way to specify a HTTPS endpoint. What happens is that given a command of:
```
swarm join --discovery consul://10.100.199.200/swarm --addr=https://dockerhost05:2376
```
the actual value that is stored in Consul's key/value store is:
```
/swarm/https:/<this is a dir>/dockerhost05:2376
```

The way I have seen this type of setup work effectively is by using an actual value in addition to the path. For example, the structure in Consul for the above `swarm join` command could be:
```
/swarm/dockerhost05, value: {  'protocol':'ssl', 'hostname':'dockerhost05', 'port':'2376' }
```

This is how VulcanD [1] is storing their info in EtcD.

[1] https://www.vulcanproxy.com/quickstart.html#example-setting-up-vulcand

**Comment by vieux** [comment]
Thanks for the report, as I was saying here https://github.com/docker/swarm/pull/208#issuecomment-68411270

I would like swarm to handle only ip:ports, no scheme or protocol.

Swarm will use https if you provide a `--tls` otherwise, regular http

**Comment by technolo-g** [comment]
I'm cool with that approach :+1:

**Comment by vieux** [comment]
Closed by #208
